src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -51,6 +56,36 @@ public Optional<Long> getDataVersion() {
     return dataVersion;
   }
 
+  /**
+   * Convenience utility for getting all attributes in Json string format.
+   *
+   * @return all attributes in Json string format., or empty string if not found.
+   */
+  public String getAttributesJson() {
+    JSONObject combinedObject = new JSONObject();
+    Stream<JSONObject> attributeJsons =
+        attributes.values().stream().map(it -> it.getJson()).filter(it -> it != null);
+
+    attributeJsons.forEach(
```

Comment:
maybe shorter:

```kt
for (entry in attributeJsons.flatMap { it.entries() }) {
   if (combinedObject.has(entry.key)) {
      Log.e(TAG, "Conflicting attribute tag Id is found: " + entry.key);
      continue;
   }

   try {
      combinedObject.put(entry.key, entry.value);
   } catch (JSONException ex) {
      Log.e(TAG, "receive attribute json exception: " + ex);
   }
}
```

assuming this works, it seems a lot shorter. Functional style seems to create a lot of indents.

Reply:
offline resolved, JSONObject is limited, there is no entries. thanks

Reply:
just update with for loop, and reduce the intent,thanks